src/iotjs_def.h

```diff
@@ -73,12 +73,72 @@
 #endif
 
 
+#ifndef __cplusplus
+typedef enum { false, true } bool;
+#endif
+
+
+#define IOTJS_VALID_MAGIC_SEQUENCE 0xfee1c001 /* feel cool */
+#define IOTJS_INVALID_MAGIC_SEQUENCE 0xfee1badd /* feel bad */
+
+#define IOTJS_DECLARE_THIS(iotjs_classname_t, x) \
+  iotjs_classname_t##_impl_t* this = &(x)->_unsafe;
+
+
+#ifdef NDEBUG
+
+#define IOTJS_VALIDATED_STRUCT(iotjs_classname_t) \
+iotjs_classname_t##_impl_t; \
+typedef struct { \
+  iotjs_classname_t##_impl_t _unsafe; \
+} iotjs_classname_t;
+
+#define IOTJS_VALIDATED_STRUCT_CONSTRUCTOR(iotjs_classname_t, x) \
+  IOTJS_DECLARE_THIS(iotjs_classname_t, x);
+#define IOTJS_VALIDATED_STRUCT_DESTRUCTOR(iotjs_classname_t, x) \
+  IOTJS_DECLARE_THIS(iotjs_classname_t, x);
+#define IOTJS_VALIDATED_STRUCT_METHOD(iotjs_classname_t, x) \
+  IOTJS_DECLARE_THIS(iotjs_classname_t, x);
+
+#else
+
+#define IOTJS_VALIDATED_STRUCT(iotjs_classname_t) \
+iotjs_classname_t##_impl_t; \
+typedef struct { \
+  unsigned _flag_create; \
+  iotjs_classname_t##_impl_t _unsafe; \
+} iotjs_classname_t;
```

Comment:
No, it cannot be used if the first member is equal to IOTJS_INVALID_MAGIC_SEQUENCE. Ok, it cannot be done.